src/libmongoc/src/mongoc/mongoc-topology-description.c

```diff
@@ -352,6 +352,10 @@ _mongoc_topology_description_server_is_candidate (
       default:
          return false;
       }
+
+   case MONGOC_TOPOLOGY_LOAD_BALANCED:
```

Comment:
Note, there is currently no call path to this function. But I added a case to silence compiler warnings on an unhandled enum value.

Load balanced server selection starts in `_mongoc_topology_select_server_id_loadbalanced` => `mongoc_topology_description_select` => `mongoc_topology_description_suitable_servers`.

I alternatively considered making this a `BSON_ASSERT (false)` since this is untested. But it seemed low impact, and this is the behavior I'd expect if `_mongoc_topology_description_server_is_candidate` was called on a load balancer topology type.

Reply:
A brief comment to that effect in the code might be nice, as this GitHub context won't easily be available to the reader.